src/main/java/edu/nus/duke/command/Duke.java

```diff
@@ -0,0 +1,121 @@
+package edu.nus.duke.command;
+
+import java.util.Scanner;
+import edu.nus.duke.task.Task;
+import edu.nus.duke.task.Todo;
+import edu.nus.duke.task.Deadline;
+import edu.nus.duke.task.Event;
+import edu.nus.duke.exception.DukeException;
+
+public class Duke {
+    // Variables
+    static final String HORIZ_LINE = "____________________________________________________________";
+    static final String CMD_TODO = "todo";
+    static final String CMD_DEADLINE = "deadline";
+    static final String CMD_EVENT = "event";
+    static Task[] tasks = new Task[100];
+
+    // Methods
+    private static void initApp() {
+        System.out.println("Hello! I'm Jarvis");
+        System.out.println("What can I do for you?");
+        System.out.println(HORIZ_LINE);
+    }
+
+    private static void finaliseApp() {
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(HORIZ_LINE);
+    }
+
+    private static void printList(int lastIndex) {
+        System.out.println("Here are the tasks in your list:");
+        for (int j = 0; j < lastIndex; j++) {
+            System.out.println((j + 1) + ". " + tasks[j].getTask());
+        }
+    }
+
+    private static boolean isAddTask(String inputTxt) {
+        return (inputTxt.startsWith(CMD_TODO) || inputTxt.startsWith(CMD_DEADLINE) || inputTxt.startsWith(CMD_EVENT));
+    }
+
+    private static void addTask(String inputTxt, int lastIndex) throws DukeException, ArrayIndexOutOfBoundsException {
+        if (inputTxt.split(" ").length == 1) {
+            throw new DukeException();
+        }
+
+        if (inputTxt.startsWith(CMD_TODO)) {
+            tasks[lastIndex] = new Todo(inputTxt.substring(5));
+        } else if (inputTxt.startsWith(CMD_DEADLINE)) {
+            String[] deadlineSplit = inputTxt.substring(9).split("/by");
+            tasks[lastIndex] = new Deadline(deadlineSplit[0].trim(), deadlineSplit[1].trim());
+        } else if (inputTxt.startsWith(CMD_EVENT)) {
+            String[] eventSplit = inputTxt.substring(6).split("/at");
+            tasks[lastIndex] = new Event(eventSplit[0].trim(), eventSplit[1].trim());
+        }
+        System.out.println("added: " + tasks[lastIndex].getTask());
+    }
+
+    private static void setDone(String inputTxt) throws ArrayIndexOutOfBoundsException, DukeException {
+        int idx = Integer.parseInt(inputTxt.split(" ")[1]) - 1;
+        if (tasks[idx] == null) {
+            throw new DukeException();
```

Comment:
It would be good if you can put an error message to `DukeException()`, so that it would be easier to debug.

Reply:
Thanks for the comment, my error message is placed outside the function using try-catch block. Do you mean 'comment' for this context?